src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -20,7 +20,26 @@ export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
       return state > value;
     case "!=":
       return state !== value;
+    case "in":
+      if (Array.isArray(state) || typeof state === typeof "string") {
+        return state.includes(value);
+      }
+      return false;
+    case "not in":
+      if (Array.isArray(state) || typeof state === typeof "string") {
+        return !state.includes(value);
+      }
+      return false;
     case "regex": {
+      if (
+        typeof state !== typeof "string" &&
+        typeof state !== typeof 1 &&
+        typeof state !== typeof true &&
+        state !== null &&
+        typeof state !== typeof undefined
+      ) {
+        return (JSON.stringify(state) as any).match(value);
+      }
       return state.match(value);
```

Comment:
This should have been test instead of match `regexp.test(string)`

Reply:
I was just matching what was already there. Is this a requirement?

Reply:
@bramkragten Also, in the original code they do not define the value variable as a regex object.

Reply:
It is a way more efficient to do a test instead of a match, so the old code was wrong.

Reply:
Okay, I will change it in another commit.

Reply:
@bramkragten I have a new commit you can check out.